flutter/news/lib/features/frontpage/presentation/bloc/articles_cubit.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:news/features/frontpage/data/repositories/articles_repository.dart';
+
+import '../../domain/entities/article.dart';
+
+part 'top_headlines_state.dart';
+
+class ArticlesCubit extends Cubit<ArticlesState> {
+  final ArticlesRepository repository;
+
+  ArticlesCubit({required this.repository}) : super(TopHeadlinesInitial());
+
+  void getTopHeadlines() async {
+    try {
```

Comment:
Do we need the `try-catch`? Could something throw here? My understanding is that `repository` will never be throwing as we are doing a proper error handling there

Reply:
Indeed its not necessary,
removed on 11abdd537712cd30e5bd412e1657e6fb5f8222ab